arangod/Sharding/ShardingInfo.cpp

```diff
@@ -307,8 +308,11 @@ void ShardingInfo::toVelocyPack(VPackBuilder& result, bool translateCids) const
   result.add(StaticStrings::WriteConcern, VPackValue(_writeConcern));
   result.add(StaticStrings::MinReplicationFactor, VPackValue(_writeConcern));
 
-  if (!_distributeShardsLike.empty() && ServerState::instance()->isCoordinator()) {
-    if (translateCids) {
+  if ((!_distributeShardsLike.empty() && ServerState::instance()->isCoordinator()) ||
+      (!_distributeShardsLike.empty() && ServerState::instance()->isSingleServer() && isSmartOrSatellite)) {
```

Comment:
puh...
can we find a different solution to this?

it took me a while to wrap my brain around that isSmartOrSatellite has no meaning within this method, then just to force-include distributeShards like.

Is this actually relevant of can we remove the isCoordinator() isSingleServer() malarky and just write down distributeShardsLike whenever it is set?

Reply:
> Is this actually relevant of can we remove the isCoordinator() isSingleServer() malarky and just write down distributeShardsLike whenever it is set?

We could, but it would impact then all collections (whenever distributeShardsLike) is set. I do not know if there will be any side effects.

Reply:
@mchacki need feedback on this.